vid/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Espresso Systems (espressosys.com)
+// This file is part of the HotShot-VID library.
+
+// You should have received a copy of the MIT License
+// along with the HotShot-VID library. If not, see <https://mit-license.org/>.
+
+//! Verifiable Information Retrieval (VID).
+#![deny(missing_docs)]
+
+use displaydoc::Display;
+use jf_poseidon2::Poseidon2Error;
+use serde::{Deserialize, Serialize};
+
+pub mod avid_m;
+mod utils;
+
+/// A glorified [`bool`] that leverages compile lints to encourage the caller to
+/// use the result.
+///
+/// Intended as the return type for verification of proofs, signatures, etc.
+/// Recommended for use in the nested [`Result`] pattern: see <https://sled.rs/errors>.
+type VerificationResult = Result<(), ()>;
+
+/// The error type for `VidScheme` methods.
+#[derive(Display, Debug)]
+pub enum VidError {
+    /// invalid args: {0}
+    Argument(String),
+    /// internal error: {0}
+    Internal(anyhow::Error),
+}
+
+impl From<Poseidon2Error> for VidError {
+    fn from(err: Poseidon2Error) -> Self {
+        VidError::Internal(err.into())
+    }
+}
+
+/// Alias
+type VidResult<T> = Result<T, VidError>;
+
+/// Trait definition for a Verifiable Information Dispersal (VID) scheme.
+pub trait VidScheme {
```

Comment:
Q: we already have `trait jf_vid::VidScheme` (and used everywhere), would it be confusing to migrate? or can we try to unify the two and stablize this trait to avoid future disruption?